packages/ra-core/src/controller/usePaginationState.spec.ts

```diff
@@ -77,4 +77,38 @@ describe('usePaginationState', () => {
             perPage: 100,
         });
     });
+
+    it('should reset the current page to 1 when perPage state changes', () => {
+        const { hookValue, childrenMock } = renderHook(() =>
+            usePaginationState()
+        );
+        expect(hookValue.pagination).toEqual({ page: 1, perPage: 25 });
+
+        act(() => hookValue.setPerPage(100));
```

Comment:
I don't understand this test. You should have a state where the page isn't 1, then change the perPage, then see that the state has a page of 1. You're doing something completely different here.